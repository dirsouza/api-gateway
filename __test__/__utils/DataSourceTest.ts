import { config } from '../../src/config';
import UserAPI from '../../src/dataSources/UserAPI';

export const mocks = {
  get: jest.fn(),
  post: jest.fn(),
  delete: jest.fn(),
  put: jest.fn(),
};

export const API = {
  userAPI: new UserAPI(config),
};
