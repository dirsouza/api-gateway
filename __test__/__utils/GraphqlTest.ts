import { ApolloServer } from 'apollo-server';
import { DataSources } from 'apollo-server-core/dist/requestPipeline';
import { importSchema } from 'graphql-import';
import resolvers from '../../src/resolvers';

import { microService } from '../../src/config';
import * as dataSources from '../../src/dataSources';
import { IDataSources } from '../../src/interfaces';

export const constructTestServer = () => {
  const apis = {
    userApi: new dataSources.UserAPI(microService)
  };

  const server = new ApolloServer({
    typeDefs: importSchema(`${__dirname}/../../src/schemas/index.graphql`),
    resolvers,
    dataSources: (): DataSources<IDataSources> => apis,
    context: () => ({})
  });

  return { server, ...apis };
};
