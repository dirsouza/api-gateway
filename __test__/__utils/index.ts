export * from './DataSourceTest';
export * from './GraphqlTest';
