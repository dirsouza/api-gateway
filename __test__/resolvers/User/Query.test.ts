import { gql } from 'apollo-server';
import { createTestClient } from 'apollo-server-testing';
import { constructTestServer } from '../../__utils';

const GET_USERS = gql(`
  query getUsers {
    getUsers {
      id
      name
      email
    }
  }
`);

const mockGetUsersResponse = {
  success: {
    success: true,
    data: [{ id: '1', name: 'Teste', email: 'teste@teste.com.br' }],
  },
  errors: {
    success: false,
    error: 'Users not found',
  },
};

describe('Teste Usuarios Resolver', () => {
  it('return all users', async () => {
    const { server, userApi } = constructTestServer();

    // @ts-ignore
    userApi.get = jest.fn(() => mockGetUsersResponse.success);

    const { query } = createTestClient(server);

    const res = await query({
      query: GET_USERS,
    });

    // @ts-ignore
    expect(res).toMatchSnapshot();
  });
});
