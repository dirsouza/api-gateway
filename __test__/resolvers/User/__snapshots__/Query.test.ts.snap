// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste Usuarios Resolver return all users (Exception) 1`] = `
Object {
  "data": Object {
    "getUsers": Array [
      Object {
        "email": "fulano@fulano.com.br",
        "id": "1",
        "name": "Fulano",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Teste Usuarios Resolver return all users 1`] = `
Object {
  "data": Object {
    "getUsers": Array [
      Object {
        "email": "fulano@fulano.com.br",
        "id": "1",
        "name": "Fulano",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
